src/geometry/manifold/mfderiv.lean

```diff
@@ -0,0 +1,1604 @@
+bring it into a convenient form to apply derivative lemmas. -/
+lemma ext_chart_preimage_inter_eq : ((ext_chart_at I x).inv_fun ⁻¹' (s ∩ t) ∩ range I.to_fun)
+  = ((ext_chart_at I x).inv_fun ⁻¹' s ∩ range (I.to_fun))
+    ∩ ((ext_chart_at I x).inv_fun ⁻¹' t) :=
+begin
+  rw [preimage_inter, inter_assoc, inter_assoc],
+  congr' 1,
+  rw inter_comm
+end
+
+end extended_charts
+
+/-- In the case of the manifold structure on a vector space, the extended charts are just the
+identity.-/
+@[simp] lemma ext_chart_model_space_eq_id (𝕜 : Type*) [nondiscrete_normed_field 𝕜]
+  {E : Type*} [normed_group E] [normed_space 𝕜 E] (x : E) :
+  ext_chart_at (model_with_corners_self 𝕜 E) x = local_equiv.refl E :=
+by simp [ext_chart_at]
+
+section derivatives_definitions
+/-!
+### Derivative of maps between manifolds
+
+The derivative of a smooth map `f` between smooth manifold `M` and `M'` at `x` is a bounded linear
+map from the tangent space to `M` at `x`, to the tangent space to `M'` at `f x`. Since we defined
+the tangent space using one specific chart, the formula for the derivative is written in terms of
+this specific chart.
+
+We use the names `mdifferentiable` and `mfderiv`, where the prefix letter `m` means "manifold".
+-/
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+{E : Type*} [normed_group E] [normed_space 𝕜 E]
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+{M : Type*} [topological_space M] [manifold H M]
+{E' : Type*} [normed_group E'] [normed_space 𝕜 E']
+{H' : Type*} [topological_space H'] (I' : model_with_corners 𝕜 E' H')
+{M' : Type*} [topological_space M'] [manifold H' M']
+
+include I
+
+/-- Predicate ensuring that, at a point and within a set, a function can have at most one
+derivative. This is expressed using the preferred chart at the considered point. -/
+def unique_mdiff_within_at (s : set M) (x : M) :=
+unique_diff_within_at 𝕜 ((ext_chart_at I x).inv_fun ⁻¹' s ∩ range I.to_fun)
+  ((ext_chart_at I x).to_fun x)
+
+/-- Predicate ensuring that, at all points of a set, a function can have at most one derivative. -/
+def unique_mdiff_on (s : set M) :=
+∀x∈s, unique_mdiff_within_at I s x
+
+include I'
+
+/-- Conjugating a function to write it in the preferred charts around `x`. The manifold derivative
+of `f` will just be the derivative of this conjugated function. -/
+def written_in_ext_chart_at (x : M) (f : M → M') : E → E' :=
+(ext_chart_at I' (f x)).to_fun ∘ f ∘ (ext_chart_at I x).inv_fun
+
+/-- `mdifferentiable_within_at I I' f s x` indicates that the function `f` between manifolds
+has a derivative at the point `x` within the set `s`.
+This is a generalization of `mdifferentiable_within_at` to manifolds. -/
+def mdifferentiable_within_at (f : M → M') (s : set M) (x : M) :=
+continuous_within_at f s x ∧
+differentiable_within_at 𝕜 (written_in_ext_chart_at I I' x f)
+  ((ext_chart_at I x).inv_fun ⁻¹' s ∩ range I.to_fun) ((ext_chart_at I x).to_fun x)
```

Comment:
Could you please explain why  differentiability does not imply continuity in case of manifolds?

Reply:
Done